packages/editor/src/plugins/fill-in-the-blanks-exercise/blank-renderer.tsx

```diff
@@ -1,133 +1,122 @@
-    // blankStates was not provided by FillInTheBlanksRenderer -> cannot continue
-    return null
-  }
-  const feedbackForBlanks = fillInTheBlanksContext.feedbackForBlanks
-  const mode = props.forceMode ?? fillInTheBlanksContext.mode
+}
 
-  const isAnswerCorrect = feedbackForBlanks.get(props.blankId)?.isCorrect
+export function BlankRenderer(props: BlankRendererProps) {
+  const { blankId, onChange } = props
 
-  const textInBlank =
-    fillInTheBlanksContext.textInBlanks.get(props.blankId)?.text ?? ''
+  const editor = useSlate()
+  const selected = useSelected()
+  const focused = useFocused()
 
-  const draggableSolutionInBlank = [
-    ...fillInTheBlanksContext.locationOfDraggables.value,
-  ].find((entry) => entry[1] === props.blankId)
+  // Autofocus when adding and removing a blank
+  const inputRef = createRef<HTMLInputElement>()
+  useEffect(() => {
+    // Focus input when the blank is added
+    const input = inputRef.current
+    if (input) input.focus()
 
-  const draggableIdInThisBlank = draggableSolutionInBlank
-    ? draggableSolutionInBlank[0]
-    : null
+    // Focus editor when the blank is removed
+    return () => {
+      ReactEditor.focus(editor)
+    }
 
-  const draggableText = fillInTheBlanksContext.draggables.find(
-    (draggable) => draggable.draggableId === draggableIdInThisBlank
-  )?.text
+    // Only run on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
 
-  return (
-    <>
-      {mode === 'typing' ? (
-        <input
-          className={cn(
-            'h-[25px] resize-none rounded-full border border-brand bg-brand-50 pl-2 pr-1',
-            isAnswerCorrect && 'border-green-500',
-            isAnswerCorrect === false && 'border-red-500'
-          )}
-          size={(textInBlank.length ?? 4) + 1}
-          spellCheck={false}
-          autoCorrect="off"
-          placeholder=""
-          type="text"
-          value={textInBlank}
-          onChange={(e) => {
-            setTextUserTypedIntoBlank(e.target.value)
-            if (props.onChange) {
-              props.onChange(e)
-            }
-          }}
-        />
-      ) : (
-        <>
-          <DroppableBlank
-            blankId={props.blankId}
-            disable={draggableIdInThisBlank !== null}
-          >
-            {draggableIdInThisBlank ? (
-              <DraggableSolution
-                text={draggableText ?? ''}
-                draggableId={draggableIdInThisBlank}
-              />
-            ) : null}
-          </DroppableBlank>
-        </>
-      )}
-    </>
-  )
+  // Focus input when the blank is selected using arrow keys
+  useEffect(() => {
+    function handleDocumentKeydown() {
+      const input = inputRef.current
+      const shouldFocusInput =
+        input &&
+        document.activeElement !== input &&
+        focused &&
+        selected &&
+        editor.selection &&
+        Range.isCollapsed(editor.selection)
+      if (shouldFocusInput) input.focus()
```

Comment:
Nice suggestion, I'll try to implement it.

For me, in Chrome, it actually does function like that, most of the time. But after your comment, I did notice inconsistencies, sometimes it will be at the end when coming from the left, so I guess that needs to be manually handled.

Reply:
Done :)